src/main/java/net/snowflake/hivemetastoreconnector/SnowflakeConf.java

```diff
@@ -130,6 +130,34 @@ public boolean isSnowflakeJDBCProperty()
 
   private static URL snowflakeConfigUrl;
 
+  /**
+   * Convenience method. Retrieves a secret from a credential provider,
+   * configuration, or default value in the respective order of preference.
+   * @param name the name of the configuration
+   * @param defaultValue the default value of the secret, if any
+   * @return the secret
+   */
+  public String getSecret(String name, String defaultValue)
+  {
+    try
+    {
+      char[] secret = getPasswordFromCredentialProviders(name);
```

Comment:
where is this defined?

Reply:
it's a method from https://hadoop.apache.org/docs/r2.7.7/api/org/apache/hadoop/conf/Configuration.html